src/main/webapp/dev/js/common/instructor.js

```diff
@@ -358,14 +358,18 @@ function bindCourseDeleteAllLinks() {
         event.preventDefault();
 
         const $clickedLink = $(event.currentTarget);
-        const messageText = 'Are you sure you want to permanently delete all the courses? '
-                + 'This operation will delete all students and sessions in these courses. '
-                + 'All instructors of these courses will not be able to access them hereafter as well.';
+        const softDeletedCourses = $('[data-soft-deleted-course-id]').toArray();
+        const messageText = `<p>Are you sure you want to permanently delete the following ${softDeletedCourses.length} `
```

Comment:
This is no longer text, we can name it as `message` though understandably not your fault because our wrapper has the parameter as `messageText`.

As a side point, we can give more meaning to the variable if we don't just name it as `message` but in this case it's entirely optional and might not be suitable either.